### PR TITLE
Add hotkey for sending readline text to editor

### PR DESCRIPTION
**Pull request by MarcoPolo-PasTonMolo**
Add a hotkey for sending text in readline to external text editor.
Should deprecate the need for /command-editor requests like /subject-editor, /correct-editor.

Example workflow:
alt+d to open editor or write something in readline first and then alt+d.

I have a feeling a put the declaration of _get_message_from_editor in a suboptimal header file.

### CODE REVIEW

**Comment by jubalh** [comment]
Thanks for this PR! Looks like a good idea!
I'll do a review soon. For now:
* can you tell us in this PR how/what you tested?
*  `_get_message_from_editor`  -> the `_` indicates that it's a local helper function and not exported.

**Comment by MarcoPolo-PasTonMolo** [comment]
1. Just wrote random stuff in readline(including commands) and then pressed alt+d. Seems to work fine.
2. Ok I'll remove the _ in a minute

**Comment by jubalh** [comment]
About testing I meant:
* Did you test with a command that requires no argument
* Requires one argument
* requires several
* commands that maybe not expect a newline? Can we think of something like this?

And I think we should move the `get_message_from_editor()` out of `cmd_funcs.c` maybe into `src/common.c`. I'm not sure what's the best location but probably that's a better one?
(@sjaeckel any opinion on this?)

**Review by sjaeckel — APPROVED** [review]
I tried it and it works well

I also think we should move `get_message_from_editor()` to another file.

**Comment by jubalh** [comment]
We need to be careful with these hotkeys. In the past we already used some that are actually used by readline per default too.

Someone collected them here: https://github.com/profanity-im/profanity/issues/1580#issuecomment-908866258

**Review by sjaeckel — COMMENTED** [review]
I'm not sure where this comes from but if I already have text written, then press `Alt+d` and then immediately close the editor, the last character is lost!

**Comment by jubalh** [comment]
Please move `get_message_from_editor()` to `src/tools/editor.{c,h}`.

**Comment by sjaeckel** [comment]
> I'm not sure where this comes from but if I already have text written, then press `Alt+d` and then immediately close the editor, the last character is lost!

I've a preliminary patch for this issue and will open a separate PR

**Comment by MarcoPolo-PasTonMolo** [comment]
> So alt + d is on there. I was curious anyways why you wanted to use d for editor? :-)
It was a mistake, wanted to have e for editor but accidentaly typed d and it stuck. Should I change it?

> Please move get_message_from_editor() to src/tools/editor.{c,h}.
Will do!

> I've a preliminary patch for this issue and will open a separate PR
Ok

**Comment by MarcoPolo-PasTonMolo** [comment]
Minor issue: what should I put after Copyright?

I saw that other files in that (tools) folder include a comment about the GNU license and someone's name above it
>  Copyright (C) 2012 - 2019 James Booth <boothj5@gmail.com>

**Comment by MarcoPolo-PasTonMolo** [comment]
I noticed some memory gets lost when the get_message_from editor command gets called, as I understand it's the returned_message that gets generated from g_strdup, I tried calling gfree on it in inputwin.c but it still gets lost

File: src/tools/editor.c [hunk 1]
```diff
@@ -110,13 +111,16 @@ get_message_from_editor(gchar* message, gchar** returned_message)
             GString* text = g_string_new(buf);
             *returned_message = g_strdup(text->str);
             g_string_free(text, TRUE);
+        } else {
+            *returned_message = g_strdup("");
```
Comment (jubalh) [inline src/tools/editor.c:115]: thats also a solution but i would prefer another one. but lets take it for now!

**Review by jubalh — APPROVED** [review]

**Comment by jubalh** [comment]
@MarcoPolo-PasTonMolo quite a cool feature! Thanks for the idea and implementation!

I think this will be a game changer for me like alt+a :D

**Comment by sjaeckel** [comment]
This relates to #1521
... and maybe that's what @hund wanted?

**Comment by Hund** [comment]
> This relates to #1521 ... and maybe that's what @Hund wanted?

I'm humbled that you though of me. :) This looks like exactly what I was looking for!

**Pull request merged**